src/mina-signer/MinaSigner.ts

```diff
@@ -441,16 +410,46 @@ class Client {
    * the sum of all account updates plus the specified fee amount. If no fee is passed in, `0.001`
    * is used (according to the Mina spec) by default.
    * @param accountUpdates A list of account updates
-   * @param fee The fee per accountUpdate amount
    * @returns  The fee to be paid by the fee payer accountUpdate
    */
-  getAccountUpdateMinimumFee(
-    accountUpdates: TransactionJson.AccountUpdate[],
-    fee: number = 0.001
-  ) {
-    return accountUpdates.reduce(
-      (accumulatedFee, _) => accumulatedFee + fee,
-      0
-    );
+  getAccountUpdateMinimumFee(accountUpdates: TransactionJson.AccountUpdate[]) {
+    return 0.001 * accountUpdates.length;
   }
 }
+
+function validNonNegative(n: number | string | bigint): string {
+  let n0 = BigInt(n); // validates that string represents an integer; also throws runtime errors for nullish inputs
+  if (n0 < 0) throw Error('input must be non-negative');
+  return n0.toString();
+}
+
+function validCommon(common: Json.Common): Json.StrictCommon {
+  return {
+    to: common.to,
```

Comment:
My understanding is that to and from are already checked base base58 decode/encode